c2corg_ui/views/document.py

```diff
@@ -432,12 +441,17 @@ def _redirect_to_full_url(self):
             raise HTTPInternalServerError(
                 "An error occurred while loading the document")
 
-        locale = data['locales'][0]
-        title = ''
-        if self._API_ROUTE == 'routes' and locale['title_prefix']:
-            title += locale['title_prefix'] + ' '
-        title += locale['title']
-        location = self.request.route_url(
-            self._API_ROUTE + '_view', id=data['document_id'],
-            lang=locale['lang'], slug=slugify(title))
-        raise HTTPMovedPermanently(location=location)
+        if 'redirects_to' in data:
+            if lang not in data['available_langs']:
+                lang = self._get_preferred_lang()
```

Comment:
Instead of using `prefered_lang`, you would have to select the *best* lang in `data['available_langs']`. Otherwise you might do a redirect to a locale that does not exist (and at this point you already know that it does not exist).